### ISSUE TITLE
set -u causes examples to fail

### ISSUE THREAD

**Issue by adudek** [niieani/bash-oo-framework#65]
tested in bash 4.3+
when setting 'set -u' in example/array.sh, I receive:
bash-oo-framework/lib/util/command.sh: line 34: $3: unbound variable

there are more errors elsewhere, after performing "${unboundvar:-}"

Is this conscious design decision related to bash version?
Sometimes I want my script to fail, not to cause more mayhem in unpredictable way. How to overcome this?

**Comment by adudek** [comment]
Note for future: "most sane" bash settings that I recommend should include at least:
set +H -euo pipefail (-e should be suppressed only in try block)
it is a pain to handle all potentially unassigned values, but less than potentially destroying data :)

**Comment by niieani** [comment]
Yes, I totally agree @adudek. We'll take this into consideration when planning 3.0. Thanks for your feedback!
